app/models/event.rb

```diff
@@ -366,6 +368,14 @@ class Event < ApplicationRecord
 
   after_update :generate_stripe_card_designs, if: -> { attachment_changes["stripe_card_logo"].present? && stripe_card_logo.attached? && !Rails.env.test? }
 
+  # We can't do this through a normal dependent: :destroy since ActiveRecord does not support deleting records through indirect has_many associations
+  # https://github.com/rails/rails/commit/05bcb8cecc8573f28ad080839233b4bb9ace07be
+  after_destroy_commit do
+    all_organizer_positions.each do |position|
```

Comment:
Any reason we couldn't do:
```suggestion
    organizer_positions.with_deleted.each do |position|
```

Reply:
that works!